src/modules/xworkspaces.cpp

```diff
@@ -197,7 +203,7 @@ namespace modules {
      * We use this to map workspaces to viewports, desktop i is at position
      * ws_positions[i].
      */
-    vector<position> ws_positions = ewmh_util::get_desktop_viewports();
+    vector<position> ws_positions = (!m_group_by_monitor) ? ewmh_util::get_desktop_viewports() : vector<position>();
```

Comment:
Shouldn't this be the other way around?

In order for grouping by monitor to work, we need the `_NET_DESKTOP_VIEWPORT` hint.

Reply:
Oh, indeed! Sorry about that.